### PR TITLE
TDD testing

### PR DESCRIPTION
**Pull request by rootDEV2990**
Please help us review our Rspec project. We have compared our methods with the Ruby ones and got greens. =)

-Thank you 🙏🏽

### CODE REVIEW

**Comment by lekegitrepo** [comment]
#### Hi @rootDEV2990 and @chindrop :clap: :clap:
#### Good job so far, however, you still have some requirement to implement before this project can be approved.

**_1. Fix the Linter error._**

**_2. Add more tests to cover all the expected behavior of each method in a test case that covers positive and negative scenarios ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/commonsense.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))._**
**_Note: 7 tests are too small to cover all the scenarios you are expected to cover or test for in this project._**

**_Optional: Fix the broken image link in the readme._**

**_Implement the required changes and submit for review._**

#### _Reviewed by Temitayo Adeleke._

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
#### Status: Request changes

**Comment by chindrop** [comment]
Changes required after reviewed made and linter errors also were corrected

File: spec/myenumerables_spec.rb [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+  describe '.my_each' do
+    it 'Makes self into array unless block is given' do
+      expect(arr.my_each).to be_a(Enumerable)
+    end
+    it 'if block is given' do
+      result = []
+      arr.my_each { |item| result << item }
+      expect(arr).to eq(result)
+    end
```
Comment (maelfosso) [inline spec/myenumerables_spec.rb]: I tweak that a bit and i got an error
```rb
expect(arr.my_each { |item| item % 2 == 0 }).to eq(arr.each { |item| item % 2 == 0 })
```

File: spec/myenumerables_spec.rb [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  # Inject needed to done again
+  describe '.my_inject' do
+    it 'it will pass each element and accumulate each sequentially' do
+      sum = query
+      for element in arr do
+        sum = sum.public_send(query_symbol, element)
+      end
+      expect(sum).to eq(arr.inject(query, query_symbol))
+    end
+  end
```
Comment (maelfosso) [inline spec/myenumerables_spec.rb]: THat test is not sufficient. Add more test here.

Also that test failed
```rb
expect(arr.my_inject { |sum, n| sum + n }).to eq arr.inject { |sum, n| sum + n }
```

File: spec/myenumerables_spec.rb [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+      expect(result).to eq(result_ruby)
+    end
+  end
+  # Inject needed to done again
+  describe '.my_inject' do
+    it 'it will pass each element and accumulate each sequentially' do
+      sum = query
+      for element in arr do
+        sum = sum.public_send(query_symbol, element)
+      end
```
Comment (maelfosso) [inline spec/myenumerables_spec.rb]: Why this?? Why don't you do thing simpler?

File: spec/myenumerables_spec.rb [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+        hash[item] = if item == query
+                       1
+                     else
+                       0
+                     end
+      end
+      result = hash.count { |x| x[1] == 1 }
+      expect(result).to eq(arr.count(query))
+    end
+  end
```
Comment (maelfosso) [inline spec/myenumerables_spec.rb]: Test that function in two situations
1- Counting a specific value
2- Counting value after applying a condition block

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello,,
Thank you for applying the suggested changes.
However, you should put a bit more effort to move forward, please

:point_right: add new test for taking in consideration all the cases,

:point_right: Read the comments left below

Kindly make the suggested changes and submit a new code review request
Thanks

Mael FOSSO

File: spec/myenumerables_spec.rb (hunk 1 above)
Comment (rootDEV2990) [reply spec/myenumerables_spec.rb]: Fixed some issues in enumberables.rb. All good to go.

File: spec/myenumerables_spec.rb (hunk 4 above)
Comment (rootDEV2990) [reply spec/myenumerables_spec.rb]: Added some more test testing Parameters and another with a Block using my_count method and comparing to Ruby's.

**Comment by chindrop** [comment]
Corrections were for reviewed test and more test were added

File: spec/myenumerables_spec.rb [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+        hash[item] = if arr[item] == arr[item.next]
+                       1
+                     else
+                       0
+                     end
+      end
+      result = hash.value?(0)
+      expect(result).to be_truthy
+    end
+  end
```
Comment (kobiyoyo) [inline spec/myenumerables_spec.rb]: Let's test when:
- [x] when a class is passed as an argument
- [x] when a Regex is passed as an argument

File: spec/myenumerables_spec.rb [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+        hash[item] = if item == query
+                       1
+                     else
+                       0
+                     end
+      end
+      result = hash.value?(1)
+      expect(result).to be_truthy
+    end
+  end
```
Comment (kobiyoyo) [inline spec/myenumerables_spec.rb]: Let's test when:
- [x] when a class is passed as an argument
- [x] when a Regex is passed as an argument

**Review by kobiyoyo — COMMENTED** [review]
## Hi Team      
#### Great Job so far, but there are a few more things required. 
### General

- [x] Kindly check comments below.
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
Changes Required

File: spec/myenumerables_spec.rb [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+        hash[item] = if item == query
+                       1
+                     else
+                       0
+                     end
+      end
+      result = false if hash.value?(1)
+      expect(result).to be_falsy
+    end
+  end
```
Comment (kobiyoyo) [inline spec/myenumerables_spec.rb]: Let's test when:
- [x] when a class is passed as an argument
- [x] when a Regex is passed as an argument

**Review by kobiyoyo — CHANGES_REQUESTED** [review]

File: spec/myenumerables_spec.rb (hunk 5 above)
Comment (rootDEV2990) [reply spec/myenumerables_spec.rb]: Ok added test cases for Classes and RegEx 👍🏽

File: spec/myenumerables_spec.rb (hunk 6 above)
Comment (rootDEV2990) [reply spec/myenumerables_spec.rb]: Ok added test cases for Classes and RegEx 👍🏽

File: spec/myenumerables_spec.rb (hunk 7 above)
Comment (rootDEV2990) [reply spec/myenumerables_spec.rb]: Ok added test cases for Classes and RegEx 👍🏽

**Comment by rootDEV2990** [comment]
Please help us review again our test cases. New additions include Classes and RegEx. 🙏🏽

**Review by kobiyoyo — APPROVED** [review]
# Project Approved :thumbsup:
## Hello Team,  You have done a brilliant job with this project. :clap:
## OPTIONAL
### Lets use ruby ternary operator for `if/else`  statements
https://www.rubyguides.com/2019/10/ruby-ternary-operator/
#### You can merge once done !!!

**Comment by rootDEV2990** [comment]
Fixed optional ternary operator.
